vision_system/TargetDetect.py

```diff
@@ -38,6 +57,46 @@ def clampZero(x, thresh=0.001):
             x = 0
         return x
 
+    def detect_color(self, frame: cv2.typing.MatLike, hls: cv2.typing.MatLike, ranges: List[List[Tuple]], color: str):
+        """_summary_
+
+        Args:
+            frame (cv2.typing.MatLike): cv2 frame to process
+            hls (cv2.typing.MatLike): frame converted to hls
+            ranges (List[List[Tuple]]): lower and upper bounds for colors in hls
+        """
+        # create masks for list of upper and lower bounds
+        masks = []
+        for range in ranges:
+            mask = cv2.inRange(hls, range[0], range[1])
+            masks.append(mask)
+
+        # combine masks
+        mask = masks[0]
+        for _mask in masks:
+            mask = np.bitwise_or(mask, _mask) 
+
+        pixels = cv2.bitwise_and(frame, frame, mask=mask)
+        gray = cv2.cvtColor(pixels, cv2.COLOR_BGR2GRAY)
+
+        cnts = cv2.findContours(gray, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
+        cnts = cnts[0] if len(cnts) == 2 else cnts[1] # if an older verison of findContours()
+
+        centroids = []
+        for c in cnts:
+            x,y,w,h = cv2.boundingRect(c)
+            A = w*h
+
+            # TODO: allow configurable the maximum area threshold
+            # skip centroids with an area below this value (might be a false positive)
+            if(A < self.MINUMUM_AREA_PX):
+                continue
```

Comment:
Resolve this when you see it but just FYI I think I'm gonna move this into the clustering algorithm I'm working on. Just so we get all the data points